apps/custom-ui-floating-window-demo/components/Dialer/ActionButtons.js

```diff
@@ -0,0 +1,137 @@
+const ActionButtons = (context) => {
+  let buttons = [];
+
+  // Takes the user back to dialer by changing the state
+  let hangup = async () => {
+    context.setCallerState("listening");
+  };
+
+  let createNewContact = async () => {
+    let notes = document.getElementById("notes").value;
+    let name = document.getElementById("contact_name").value;
+    // Create Contact
+    const newContact = await fetch(`/api/addContact`, {
+      method: "POST",
+      body: JSON.stringify({
+        name,
+        number: context.callerDetails.number,
+        notes,
+      }),
+    });
+    let contactObj = await newContact.json();
+    let contact = contactObj.data;
+
+    context.setCallerDetails({ ...context.callerDetails, id: contact.id });
+    context.setCallerState("disconnected");
+  };
+
+  let addNotesToDeal = async () => {
+    const note = document.getElementById("notes").value;
+    const dealId = document.getElementById("deals").value;
+
+    const start_time = new Date()
+      .toISOString()
+      .replace(/T/, " ")
+      .replace(/\..+/, "");
+    const end_time = start_time;
+
+    const res = await fetch(`/api/addCallLogsToDeal`, {
+      method: "POST",
+      body: JSON.stringify({
+        dealId,
+        note,
+        to_phone_number: context.callerDetails.number,
+        start_time,
+        end_time,
+        outcome: "busy",
+      }),
+    });
+
+    context.setCallerState("disconnected");
+  };
+
+  let addNotesToContact = async () => {
+    let notes = document.getElementById("notes").value;
+
+    await fetch(`/api/addNotesToContact`, {
+      method: "POST",
+      body: JSON.stringify({
+        id: context.callerDetails.id,
+        notes,
+      }),
+    });
+
+    context.setCallerState("disconnected");
+  };
+
+  // Only show the 'Answer' and 'Reject' buttons during Incoming calls
+  if (
+    context.callerDetails.direction === "in" &&
+    context.callerState === "ringing"
+  ) {
+    // Answer Button
+    buttons.push(
```

Comment:
Pushing to a mutable variable is anti-pattern in react, can you refactor this component to something like this:

```
if (someCondition) {
    return (
        <>
             <button>button 1</button>
             <button>button 2</button>
        </>
    );
}

if (someOtherCondition) {
    return (
        <>
             <button>button 3</button>
             <button>button 4</button>
        </>
    );
}
```